src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/BalanceCalculator.java

```diff
@@ -39,22 +34,19 @@ public class BalanceCalculator {
       "Agreement record not found for tenantId {0}, personId {1} and date {2}";
   static final String ACCRUALS_NOT_FOUND =
       "No Accrual records found for tenantId {0} and personId {1} between {2} and {3}";
-  static final String MISSING_ACCRUAL =
-      "Accrual missing for tenantId {0}, personId {1}, accrual type {2} and date {3}";
-  static final String ACCRUALS_MAP_EMPTY = "Accruals Map must contain at least one entry!";
 
   private final RestClient restClient;
-  private final List<AccrualModule> accrualModules;
+  private final ContributionsHandler contributionsHandler;
 
   @Autowired
-  public BalanceCalculator(RestClient restClient, List<AccrualModule> accrualModules) {
+  public BalanceCalculator(RestClient restClient, ContributionsHandler contributionsHandler) {
     this.restClient = restClient;
-    this.accrualModules = accrualModules;
+    this.contributionsHandler = contributionsHandler;
   }
 
-  public List<Accrual> calculate(TimeEntry timeEntry) {
+  public List<Accrual> calculate(TimeEntry timeEntry, KafkaAction action) {
 
-    String timeEntryId = timeEntry.getId();
+    // TODO: what to do if TimeEntry has no end date ?? / exit ??
```

Comment:
Is there a story to cover this scenario or do we need to speak to the designers to make sure we get it covered?

Reply:
We believe we have an idea of how this would look. But it is a question we are gonna take to the designers to make sure our thoughts are correct.